vsphere/resource_vsphere_host.go

```diff
@@ -826,15 +836,77 @@ func hostLockdownString(lockdownMode types.HostLockdownMode) (string, error) {
 	return "", fmt.Errorf("unknown Lockdown mode encountered")
 }
 
-func buildHostConnectSpec(d *schema.ResourceData) types.HostConnectSpec {
+func buildHostConnectSpec(d *schema.ResourceData) (types.HostConnectSpec, error) {
+	thumbprint := d.Get("thumbprint").(string)
+	hostname := d.Get("hostname").(string)
+	username := d.Get("username").(string)
+	password := d.Get("password").(string)
+
+	log.Printf("Building HostConnectSpec for host: %s", hostname)
+	// Retrieve the actual thumbprint from the host
+	actualThumbprint, err := getHostThumbprint(d)
+	if err != nil {
+		log.Printf("Error retrieving host thumbprint: %s", err)
+		return types.HostConnectSpec{}, fmt.Errorf("error retrieving host thumbprint: %s", err)
+	}
+
+	// Compare the retrieved thumbprint with the provided thumbprint
+	if thumbprint != actualThumbprint {
+		log.Printf("Thumbprint mismatch: expected %s, got %s", thumbprint, actualThumbprint)
+		return types.HostConnectSpec{}, fmt.Errorf("thumbprint mismatch: expected %s, got %s", thumbprint, actualThumbprint)
+	}
+
 	hcs := types.HostConnectSpec{
-		HostName:      d.Get("hostname").(string),
-		UserName:      d.Get("username").(string),
-		Password:      d.Get("password").(string),
-		SslThumbprint: d.Get("thumbprint").(string),
+		HostName:      hostname,
+		UserName:      username,
+		Password:      password,
+		SslThumbprint: thumbprint,
 		Force:         d.Get("force").(bool),
 	}
-	return hcs
+	return hcs, nil
+}
+
+func getHostThumbprint(d *schema.ResourceData) (string, error) {
+	config := &tls.Config{}
+
+	// Check and handle the "hostname" field
+	address, ok := d.Get("hostname").(string)
+	if !ok {
+		return "", fmt.Errorf("hostname field is not a string or is nil")
+	}
+
+	// Default port for HTTPS
+	port := "443"
+	if p, ok := d.GetOk("port"); ok {
+		if portStr, ok := p.(string); ok {
+			port = portStr
+		} else {
+			return "", fmt.Errorf("port field is not a string")
+		}
+	}
+
+	config = &tls.Config{
+		InsecureSkipVerify: true,
```

Comment:
I think this should be based on `insecure_connection`.

Reply:
walk me through your thoughts on this

Reply:
```suggestion
		InsecureSkipVerify: d.Get("insecure").(bool),
```

Reply:
The obove suggestion will set the value based on the provider configuration.

This is also done in `data_source_vsphere_host_thumbprint.go` but like this, which is also valid.

```go
	config := &tls.Config{}
	config.InsecureSkipVerify = d.Get("insecure").(bool)
```

Reply:
i thought that was the idea, i had to change the datasource and also the documentations to make insecure required